lib/windows/app/index.js

```diff
@@ -67,6 +67,15 @@ const params = new URL(window.location).searchParams;
 const appPath = params.get('appPath');
 const app = loadApp(appPath);
 
+function appInitializeAction(options) {
+    return {
+        type: 'APP_INITIALIZE',
+        ...options,
+    };
+}
+
 const store = configureStore(rootReducer, app);
 const rootElement = React.createElement(RootContainer, { store });
 render(rootElement, document.getElementById('webapp'));
+
+store.dispatch(appInitializeAction({ appPath }));
```

Comment:
The `render` function is synchronous at the moment, but may that could change in the future: https://facebook.github.io/react/docs/react-dom.html#reference. Then this action could potentially be dispatched before render is complete. One way to solve it would be to pass a callback function to render (ref the doc) and dispatch the action from there.

However, I would suggest dispatching the action from `Root.componentDidMount()` instead. This is invoked after `Root` and all its children have rendered for the first time. Then we could dispatch other lifecycle actions from `Root` as well, f.ex:
- `componentWillMount` could dispatch `APP_INIT`
- `componentDidMount` could dispatch `APP_READY`
- `componentWillUnmount` could dispatch `APP_EXITING`

We could have an `appLifecycleActions.js` with these types of actions, and pass them as props from `RootContainer` to `Root`. What do you think?

Also, things like `appPath` could be useful in other places as well, so we could pass it as part of the `api` object to the action creators. Then we do not need to include it with the lifecycle actions.